### ISSUE TITLE
self-assigning-variable false positive on tuple unpacking

### ISSUE THREAD

**Issue by conqp** [PyCQA/pylint#3433]
### Steps to reproduce
Use one-element-tuple unpacking like `var, = var`

### Current behavior
Get warning `W0127: Assigning the same variable 'var' to itself (self-assigning-variable)`

### Expected behavior
Get no warning at all.

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.1 (default, Jan 22 2020, 06:38:00) 
[GCC 9.2.0]
```

### Actual use case
Variable `previous` in the following example:
```
def check_state_change(system, check):
    """Checks the state change for the respective system and check."""

    select = check.select().where(check.system == system)
    select = select.order_by(check.timestamp.desc()).limit(2)

    try:
        last, *previous = select
    except ValueError:
        raise NotChecked(system, check)

    if previous:
        previous, = previous

        if last.successful and not previous.successful:
            return CheckState(system, last, State.RECOVERED)

        if previous.successful and not last.successful:
            return CheckState(system, last, State.FAILED)

        return CheckState(system, last, State.UNCHANGED)

    if last.successful:
        return CheckState(system, last, State.RECOVERED)

    return CheckState(system, last, State.FAILED)
```
